flyteadmin/auth/handlers.go

```diff
@@ -154,6 +154,12 @@ func GetLoginHandler(ctx context.Context, authCtx interfaces.AuthenticationConte
 				logger.Errorf(ctx, "Was not able to create a redirect cookie")
 			}
 		}
+
+		// Add the IDPQueryParameter to the URL if it is present in the request
+		idpQueryParam := authCtx.Options().UserAuth.IDPQueryParameter
+		if len(idpQueryParam) > 0 && queryParams.Get(idpQueryParam) != "" {
+			url = fmt.Sprintf("%s&%s=%s", url, idpQueryParam, queryParams.Get(idpQueryParam))
```

Comment:
can we use the url library to encode this?

url.Query.Add() https://pkg.go.dev/net/url#example-URL.Query

Reply:
Done